### PR TITLE
Issues/4005 add real time logs

### PR DESCRIPTION
**Pull request by kannon92**
## Changelog Entry

cwltool pr: https://github.com/common-workflow-language/cwltool/pull/1600

Add the ability to stream logs to a directory on the host path rather than in the container.  

## Reviewer Checklist

 * [ ] Make sure it is coming from `issues/XXXX-fix-the-thing` in the Toil repo, or from an external repo.
    * [ ] If it is coming from an external repo, make sure to pull it in for CI with:
        ```
        contrib/admin/test-pr otheruser theirbranchname issues/XXXX-fix-the-thing
        ```
    * [ ] If there is no associated issue, [create one](https://github.com/DataBiosphere/toil/issues/new).
* [ ] Read through the code changes. Make sure that it doesn't have:
    * [ ] Addition of trailing whitespace.
    * [ ] New variable or member names in `camelCase` that want to be in `snake_case`.
    * [ ] New functions without [type hints](https://docs.python.org/3/library/typing.html).
    * [ ] New functions or classes without informative docstrings.
    * [ ] Changes to semantics not reflected in the relevant docstrings.
    * [ ] New or changed command line options for Toil workflows that are not reflected in `docs/running/{cliOptions,cwl,wdl}.rst` 
    * [ ] New features without tests.
* [ ] Comment on the lines of code where problems exist with a review comment. You can shift-click the line numbers in the diff to select multiple lines.
* [ ] Finish the review with an overall description of your opinion.

## Merger Checklist

* [ ] Make sure the PR passes tests.
* [ ] Make sure the PR has been reviewed **since its last modification**. If not, review it.
* [ ] Merge with the Github "Squash and merge" feature.
    * [ ] If there are multiple authors' commits, add [Co-authored-by](https://github.blog/2018-01-29-commit-together-with-co-authors/) to give credit to all contributing authors.
* [ ] Copy its recommended changelog entry to the [Draft Changelog](https://github.com/DataBiosphere/toil/wiki/Draft-Changelog).
* [ ] Append the issue number in parentheses to the changelog entry.

### CODE REVIEW

**Review by DailyDreaming — APPROVED** [review]
LGTM.  Thank you @kannon92 .  Is this ready to merge?

**Comment by mr-c** [comment]
The `cwltool` side needs merging and release. Should also have a test on the `toil-cwl-runner` side as well

**Comment by kannon92** [comment]
> The `cwltool` side needs merging and release. Should also have a test on the `toil-cwl-runner` side as well

Thanks for calling that out!  I will add that.

**Comment by kannon92** [comment]
> The `cwltool` side needs merging and release. Should also have a test on the `toil-cwl-runner` side as well

Thank you for merging the PR for cwltool.  I'm planning on adding tests soon!

When do you expect a release of cwltool?

**Comment by mr-c** [comment]
@kannon92 https://pypi.org/project/cwltool/3.1.20220204090313/

Can you merge in the latest from https://github.com/DataBiosphere/toil ? I'll push it up for testing then

**Comment by kannon92** [comment]
> @kannon92 https://pypi.org/project/cwltool/3.1.20220204090313/
> 
> Can you merge in the latest from https://github.com/DataBiosphere/toil ? I'll push it up for testing then

Sure thing.  I am working on the test cases.  So please don't merge until I say its good to go.

File: src/toil/test/cwl/cwlTest.py [hunk 1]
```diff
@@ -660,55 +750,185 @@ def test_streamable(self):
             self.assertEqual(f.read().strip(), "When is s4 coming out?")
 
 
+@needs_cwl
```
Comment (kannon92) [inline src/toil/test/cwl/cwlTest.py:753]: The unit tests that I added start here.  

I ran black on the cwlTest.py file and I guess it wasn't formatted in a while.  

I can roll this back if you all want.

**Comment by kannon92** [comment]
> > @kannon92 https://pypi.org/project/cwltool/3.1.20220204090313/
> > Can you merge in the latest from https://github.com/DataBiosphere/toil ? I'll push it up for testing then
> 
> Sure thing. I am working on the test cases. So please don't merge until I say its good to go.

Alright.  I finally got the tests working without any fickleness.  

Feel free to pull this into CI.

**Comment by kannon92** [comment]
Let me know if there is anything else you need from me.  The CI didn't show anything alarming so its good to go on my end.  Please merge whenever you get a chance.

**Pull request merged**